vsintegration/src/FSharp.Editor/QuickInfo/QuickInfoProvider.fs

```diff
@@ -33,6 +35,17 @@ type internal FSharpQuickInfoProvider
 
     let xmlMemberIndexService = serviceProvider.GetService(typeof<SVsXMLMemberIndexService>) :?> IVsXMLMemberIndexService
     let documentationBuilder = XmlDocumentation.CreateDocumentationBuilder(xmlMemberIndexService, serviceProvider.DTE)
+    let navigateTo (workspace: Workspace) (range: range)  = async {
+        let documentNavigationService =workspace.Services.GetService<IDocumentNavigationService>()
+        let filePath = System.IO.Path.GetFullPathSafe range.FileName
+        match workspace.CurrentSolution.GetDocumentIdsWithFilePath(filePath) |> List.ofSeq with
+        | id :: _ ->
```

Comment:
This one is tricky. It seems workspace will not open another editor if we already have a file opened in IDE from the wrong project. But you're right, main concern is to not open from wrong project during navigation.
Range doesn't carry assembly info but we have the Symbol because we're in the context of quickinfo, so it's doable.